src/libs/ajax/data-table-providers/WDSDataTableProvider.ts

```diff
@@ -0,0 +1,117 @@
+import _ from 'lodash/fp'
+import { Ajax } from 'src/libs/ajax'
+import { DataTableFeatures, DataTableProvider, DeleteTableFn, DownloadTsvFn, EntityMetadata, EntityQueryOptions, EntityQueryResponse, GetMetadataFn, GetPageFn } from 'src/libs/ajax/data-table-providers/DataTableProvider'
+
+// interface definitions for WDS payload responses
+interface AttributeSchema {
+  name: string,
+  datatype: string,
+  relatesTo?: string
+}
+
+export interface RecordTypeSchema {
+  name: string,
+  count: number,
+  attributes: AttributeSchema[]
+}
+
+interface SearchRequest {
+  offset: number,
+  limit: number,
+  sort: 'asc' | 'ASC' | 'desc' | 'DESC',
+  sortAttribute: string
+}
+
+interface RecordResponse {
+  id: string,
+  type: string,
+  attributes: { [index: string]: any } // truly "any" here; the backend Java representation is Map<String, Object>
```

Comment:
I wonder if `unknown` would be preferable here, so that TS informs you that you need to figure out what type the attribute is before using it.

https://www.typescriptlang.org/docs/handbook/2/functions.html#unknown

Reply:
I'd also consider using the very convenient Record<Key, Value> type.... so:

`  attributes: Record<string, unknown>`